client/v2/client.go

```diff
@@ -61,6 +70,9 @@ type Client interface {
 	// Query makes an InfluxDB Query on the database. This will fail if using
 	// the UDP client.
 	Query(q Query) (*Response, error)
+
+	// Close releases any resources a Client may be using.
+	Close() error
```

Comment:
I'd rather not make users create their own `net.Conn`, seems overly complicated. IMO it's okay to have a `Close()` function without an `Open()`. In the `net` library itself the `Close()` function is paired with the `DialUDP()` function.

This also means that we need `NewUDPClient` to return an error, correct?